packages/tpetra/core/src/Tpetra_Details_resizeRowPtrs.hpp

```diff
@@ -0,0 +1,192 @@
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+
+#ifndef TPETRA_DETAILS_RESIZEROWPTR_HPP
+#define TPETRA_DETAILS_RESIZEROWPTR_HPP
+
+#include "TpetraCore_config.h"
+#include "Kokkos_Core.hpp"
+#include "Kokkos_UnorderedMap.hpp"
+
+/// \file Tpetra_Details_resizeRowPtr.hpp
+/// \brief Functions that resizes CSR row pointers and indices
+/// \warning This file, and its contents, are implementation details
+///   of Tpetra.  The file itself or its contents may disappear or
+///   change at any time.
+
+namespace Tpetra {
+namespace Details {
+
+namespace pad_csr_details {
+
+template<class ViewType>
+ViewType empty_view(const std::string& name, const size_t& size) {
+  ViewType v(Kokkos::view_alloc(name, Kokkos::WithoutInitializing), size);
+  return v;
+}
+
+// Determine if row_ptr and indices arrays need to be resized to accommodate
+// new entries
+template<class RowPtr, class Indices, class Padding>
+void
+pad_csr_arrays(RowPtr& row_ptr_beg, RowPtr& row_ptr_end, Indices& indices, const Padding& padding)
+{
+  using range = Kokkos::pair<typename RowPtr::value_type, typename RowPtr::value_type>;
+
+  // Determine if the indices array is large enough
+  auto num_row = row_ptr_beg.size() - 1;
```

Comment:
Could `row_ptr_beg` ever be length 0, e.g., because there are no entries in the matrix/graph on the calling process?

Reply:
Yes, in fact I had a test that did just that (inadvertently) and this function hung.  I fixed the test - I did not fix the function, yet